src/Core/Commander/Providers/WFS_Provider.js

```diff
@@ -17,50 +25,136 @@ import CacheRessource from 'Core/Commander/Providers/CacheRessource';
  * @param {String} options.format: image format (default: format/jpeg)
  * @returns {Object@call;create.url.url|String}
  */
-function WFS_Provider(options) {
-
-    this.cache = CacheRessource();
-    this.ioDriver_JSON = new IoDriver_JSON();
-    this.baseUrl = options.url || "";
-    this.layer = options.layer || "";
-    this.typename = options.typename || "";
-    this.format = defaultValue(options.format, "json");
-    this.epsgCode = options.epsgCode || 4326;
+function WFS_Provider(/*options*/) {
+    Provider.call(this, new IoDriver_JSON());
+    this.cache          = CacheRessource();
+    this.ioDriverXML    = new IoDriverXML();
+    this.projection     = new Projection();
+    this.tool           = new FeatureToolBox();
 }
 
-
 WFS_Provider.prototype = Object.create(Provider.prototype);
 WFS_Provider.prototype.constructor = WFS_Provider;
 
+WFS_Provider.prototype.url = function(coord, layer) {
+    var bbox;
+    if (layer.type == "point" || layer.type == "line" || layer.type == "box")
```

Comment:
not layer type, but data type, we must tell to WFS_Provider to select a good parser for input data. It's also defined in Deginext Doc i think.